### DISCUSSION TITLE
Help with WKB

### DISCUSSION THREAD

**Discussion by Marsevil** [visgl/loaders.gl#2962] [Q&A]
Hello everyone, I'm trying to fetch and show a WKB, so I'm using the WKBLoader and thanks to [this page](https://deck.gl/docs/api-reference/layers/geojson-layer#using-binary-data) I found a way to put my `BinaryGeometry` (here `BinaryPointGeometry` for test purpose) in a `GeoJsonLayer` which should, in my comprehension, handle that despite his name.
Unfortunately, I got the error `deck: initialization of GeoJsonLayer({id: 'wkbPoint'}): GeoJSON does not have type Error: GeoJSON does not have type` from Deck.GL.

**Note** that my `BinaryPointGeometry` actually has a `type` field with value `Point`.

Here are some sources
```typescript
const rawData = await fetch(FETCH_URL)
            .then(res => res.blob())
        const data = await parse(rawData, WKBLoader, {
            worker: false,
            wkb: {
                shape: 'binary-geometry'
            }
        }) as BinaryPointGeometry;

        console.debug("Parsed type : ", data.type);
        console.debug("Data : ", data.positions);

        const layer = new GeoJsonLayer({
            id: 'wkbPoint',
            data: {
                shape: 'binary-feature-collection',
                points: data,
            } as BinaryFeatureCollection,
            pointType: 'circle', 
            getFillColor: [255, 0, 0, 255],
            getRadius: 2000,
        })
```

**Comment by ibgreen** [comment]
Hmm, I assume you have checked the structure of `data.positions`. 

You may want to apply types using checking he shape of the returned data, and using e.g. `as const satisfies` rather than just `as` - those constructions are less likely to hide type problems.

```ts
const rawData = await fetch(FETCH_URL)
         .then(res => res.blob())
        const data = await parse(rawData, WKBLoader, {
            worker: false,
            wkb: {
                shape: 'binary-geometry'
            }
        });

     if (data.type === 'binary-geometry') {
        console.debug("Parsed type : ", data.type);
        console.debug("Data : ", data.positions);

        const layer = new GeoJsonLayer({
            id: 'wkbPoint',
            data: {
                shape: 'binary-feature-collection',
                points: data,
            } as const satisifies BinaryFeatureCollection,
            pointType: 'circle', 
            getFillColor: [255, 0, 0, 255],
            getRadius: 2000,
        })
```

I don't have a working example of WKB in loaders.gl examples, that could be a good addition. Until we get around to that if you could create a code pen that might make it a bit easier to help you debug.

**Reply by Marsevil** [reply]
Sorry for the late answer
Actually `data.type` is equal to `Point`, `LineString` or `Polygon`.
By seeking in the documentation and experimenting different things I end up with the following
```ts
const geoLayer = new GeoJsonLayer({
      id: "geo-layer",
      data: {
        type: "FeatureCollection",
        shape: "binary-feature-collection",
        features: [
          data,
        ],
      },
      getFillColor: [0, 255, 0, 255],
      getLineColor: [0, 255, 0, 255],
      getLineWidth: 2000,
      getPointRadius: 2000,
      getElevation: 10,
    });
```
Even if it does not generate errors, nothing's shown on the map :cry:

**Reply by ibgreen** [reply]
See https://github.com/visgl/loaders.gl/blob/1a8de78c340c7fb2e8a266ba552a3202872e37fb/modules/schema/src/types/binary-geometries.ts#L62

**Reply by Marsevil** [reply]
I tried that too but I've got `deck: initialization of GeoJsonLayer({id: 'geo-layer'}): GeoJSON does not have type Error: GeoJSON does not have type`

Btw I pushed my code [here](https://github.com/Marsevil/svelte-deckgl-sandbox)
And I build the layer [here](https://github.com/Marsevil/svelte-deckgl-sandbox/blob/8b5c641ce74818dc09abc25956630a44371e612d/streaming-geoarrow-webui/src/lib/layers.ts#L56)
